src/goose/cli/session.py

```diff
@@ -218,6 +218,8 @@ def reply(self) -> None:
                 for tool_use in response.tool_use:
                     tool_result = self.exchange.call_function(tool_use)
                     content.append(tool_result)
+                    if tool_use.name == "get_hints":  # todo find more elegant solution to inject hints into moderator
```

Comment:
could you help me understand what makes this not elegant? maybe include some details on the lifecycle of a hints tool?

Reply:
Feels a bit weird to make a case here for this particular tool use to then change the moderator state. Would maybe prefer if the tool could act on the moderator state directly?